GVFS/GVFS.Common/Git/GitProcess.cs

```diff
@@ -373,6 +373,16 @@ public Result UnpackObjects(Stream packFileStream)
                 null);
         }
 
+        public Result PackObjects(string filenamePrefix, string gitObjectsDirectory, Action<StreamWriter> packFileStream)
+        {
+            string packFilePath = Path.Combine(gitObjectsDirectory, "pack", filenamePrefix);
+            return this.InvokeGitAgainstDotGitFolder(
+                $"pack-objects {packFilePath} --non-empty",

```

Comment:
We may want a `-q` here to avoid progress output. But it depends on what the logs look like.